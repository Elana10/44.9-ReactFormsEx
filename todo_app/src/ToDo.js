import React, { useState } from "react";

const ToDo = ({id, todo, removeToDo}) =>{
    const handleRemove =(e) => {
        const {id} = e.target;
        removeToDo(id)
    }


    return (
        <>
            <li>
                {todo}
            </li>
            <button onClick = {handleRemove} id={id}>Remove</button>        
        </>

    )
}


export default ToDo;