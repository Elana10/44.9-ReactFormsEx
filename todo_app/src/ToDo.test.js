import React from "react";
import { render, fireEvent, getByText } from "@testing-library/react";
import ToDo from "./ToDo";

it("renders without crashing", function() {
    render(<ToDo/>)
} ) 

it("matches snapshot", function() {
    const {asFragment} = render(<ToDo/>);
    expect(asFragment()).toMatchSnapshot();
})

