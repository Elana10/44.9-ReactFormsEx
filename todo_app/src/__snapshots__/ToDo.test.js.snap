// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles remove clicks 1`] = `
<DocumentFragment>
  <li>
    Walk dog
  </li>
  <button
    id="1"
  >
    Remove
  </button>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li />
  <button>
    Remove
  </button>
</DocumentFragment>
`;
